tests/h/views/api/moderation_test.py

```diff
@@ -3,62 +3,87 @@
+        assert annotation.is_hidden
+
+    def test_it_does_not_modify_an_already_hidden_annotation(
+        self, pyramid_request, annotation, annotation_context
+    ):
+        moderation = AnnotationModeration()
+        annotation.moderation = moderation
+
+        views.create(annotation_context, pyramid_request)
+
+        assert annotation.is_hidden
+        # It's the same one not a new one
+        assert annotation.moderation == moderation
+
+    def test_it_publishes_update_event(
+        self, pyramid_request, annotation_context, events
+    ):
+        views.create(annotation_context, pyramid_request)
 
         events.AnnotationEvent.assert_called_once_with(
-            pyramid_request, resource.annotation.id, "update"
+            pyramid_request, annotation_context.annotation.id, "update"
         )
 
         pyramid_request.notify_after_commit.assert_called_once_with(
             events.AnnotationEvent.return_value
         )
 
-    def test_it_renders_no_content(self, pyramid_request, resource):
-        response = views.create(resource, pyramid_request)
+    def test_it_renders_no_content(self, pyramid_request, annotation_context):
+        response = views.create(annotation_context, pyramid_request)
         assert isinstance(response, HTTPNoContent)
 
 
-@pytest.mark.usefixtures("moderation_service")
 class TestDelete:
     def test_it_unhides_the_annotation(
-        self, pyramid_request, resource, moderation_service
+        self, pyramid_request, annotation, annotation_context
     ):
-        views.delete(resource, pyramid_request)
+        annotation.moderation = AnnotationModeration()
 
-        moderation_service.unhide.assert_called_once_with(resource.annotation)
+        views.delete(annotation_context, pyramid_request)
 
-    def test_it_publishes_update_event(self, pyramid_request, resource, events):
-        views.delete(resource, pyramid_request)
+        assert not annotation.is_hidden
+
+    def test_it_publishes_update_event(
+        self, pyramid_request, annotation_context, events
+    ):
+        views.delete(annotation_context, pyramid_request)
 
         events.AnnotationEvent.assert_called_once_with(
-            pyramid_request, resource.annotation.id, "update"
+            pyramid_request, annotation_context.annotation.id, "update"
         )
 
         pyramid_request.notify_after_commit.assert_called_once_with(
             events.AnnotationEvent.return_value
         )
 
-    def test_it_renders_no_content(self, pyramid_request, resource):
-        response = views.delete(resource, pyramid_request)
+    def test_it_renders_no_content(self, pyramid_request, annotation_context):
+        response = views.delete(annotation_context, pyramid_request)
         assert isinstance(response, HTTPNoContent)
 
 
 @pytest.fixture
-def resource():
-    return mock.Mock(spec_set=["annotation", "group"])
+def annotation(factories):
+    return factories.Annotation()
```

Comment:
Another stealth `AnnotationContext` here. These are very hard to find when trying to fix all instances of a thing for example.

I wonder if we should should go looking for these plain mocks? They are almost always a bad idea.